hazelcast-client/src/main/java/com/hazelcast/client/util/ClientDelegatingFuture.java

```diff
@@ -189,87 +177,62 @@ public V join() {
         }
     }
 
-    private Object getResult() {
+    private Object resolveResponse(ClientMessage clientMessage, boolean deserialize) {
         if (defaultValue != null) {
             return defaultValue;
         }
 
-        // If value is already deserialized, use it.
-        if (deserializedValue != null) {
-            return deserializedValue;
-        }
-        if (deserializeResponse) {
-            // Otherwise, it is possible that received data may not be deserialized
-            // if "shouldDeserializeData" flag is not true in any of registered "DelegatingExecutionCallback".
-            // So, be sure that value is deserialized before returning to caller.
-            deserializedValue = serializationService.toObject(response);
-            return deserializedValue;
+        ClientMessage tempMessage = ClientMessage.createForDecode(clientMessage.buffer(), 0);
```

Comment:
Creates a new ClientMessage object but wraps internal buffer. The buffer is not copied.